pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -1374,13 +1337,52 @@ type RulesStatus struct {
 	// TODO: add status information.
 }
 
+// CommonScrapeEndpointSpec
+type endpointParams struct {
```

Comment:
Hm, sounds like this is only for one function use (that we use twice). Perhaps we could keep ScrapeEndpoint in that function as param and kill this struct (and make sure we transition from ScrapeNodeEndpoint to ScrapeEndpoint for that one function)? Sounds like a similar choice, just less code (I understand roughly ScrapeEndpoint >= ScrapeNodeEndpoint in scope)

Reply:
Great idea!